src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/GatherTranches.java

```diff
@@ -0,0 +1,87 @@
+        programGroup = ReadProgramGroup.class
+)
+@DocumentedFeature
+public class GatherTranches extends CommandLineProgram {
+    @Argument(fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME, doc="List of scattered tranches files")
+    public final List<File> inputReports = new ArrayList<>();
+
+    /**
+     * Add truth sensitivity slices through the call set at the given values. The default values are 100.0, 99.9, 99.0, and 90.0
+     * which will result in 4 estimated tranches in the final call set: the full set of calls (100% sensitivity at the accessible
+     * sites in the truth set), a 99.9% truth sensitivity tranche, along with progressively smaller tranches at 99% and 90%.
+     */
+    @Argument(fullName="TStranche",
+            shortName="tranche",
+            doc="The levels of truth sensitivity at which to slice the data. (in percent, that is 1.0 for 1 percent)",
+            optional=true)
+    private List<Double> TS_TRANCHES = new ArrayList<>(Arrays.asList(100.0, 99.9, 99.0, 90.0));
+
+    /**
+     * Use either SNP for recalibrating only SNPs (emitting indels untouched in the output VCF) or INDEL for indels (emitting SNPs untouched in the output VCF). There is also a BOTH option for recalibrating both SNPs and indels simultaneously, but this is meant for testing purposes only and should not be used in actual analyses.
+     */
+    @Argument(fullName = "mode", shortName = "mode", doc = "Recalibration mode to employ", optional = false)
+    public VariantRecalibratorArgumentCollection.Mode MODE = VariantRecalibratorArgumentCollection.Mode.SNP;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc="File to output the gathered tranches file to")
+    public File outputReport;
+
+    private PrintStream tranchesStream;
+
+    @Override
+    protected Object doWork() {
+        inputReports.forEach(IOUtil::assertFileIsReadable);
+
+        try {
+            tranchesStream = new PrintStream(outputReport);
+        } catch (FileNotFoundException e) {
+            throw new UserException.CouldNotCreateOutputFile(outputReport, e);
+        }
+
+
+        //use a data structure to hold the tranches from each scatter shard in a format that's easy to merge
+        final Map<Double, List<VQSLODTranche>> scatteredTranches = new HashMap<>();
+        for (final File trancheFile : inputReports) {
+            try {
+                List<VQSLODTranche> shardTranches = VQSLODTranche.readTranches(trancheFile);
+                for (final VQSLODTranche currentTranche : shardTranches) {
+                    if (scatteredTranches.containsKey(currentTranche.minVQSLod)) {
+                        final List<VQSLODTranche> toAdd = scatteredTranches.get(currentTranche.minVQSLod);
+                        toAdd.add(currentTranche);
```

Comment:
I don't understand this if block.  It doesn't look like the list inside is ever used outside of it...